tokio-util/src/io/reader_stream.rs

```diff
@@ -65,6 +65,19 @@ impl<R: AsyncRead> ReaderStream<R> {
             buf: BytesMut::new(),
         }
     }
+
+    /// Convert an [`AsyncRead`] into a [`Stream`] with item type
+    /// `Result<Bytes, std::io::Error>`,
+    /// with a specific read buffer initial capacity.
+    ///
+    /// [`AsyncRead`]: tokio::io::AsyncRead
+    /// [`Stream`]: futures_core::Stream
+    pub fn with_capacity(reader: R, capacity: usize) -> Self {
+        ReaderStream {
+            reader: Some(reader),
+            buf: BytesMut::with_capacity(capacity),
```

Comment:
The capacity is also used in the `poll_next` method here:
```Rust
if this.buf.capacity() == 0 {
    this.buf.reserve(CAPACITY);
}
```
This can happen during use of a `ReaderStream` since capacity is given up when `split` is called.